drivers/SmartThings/zigbee-switch/src/zll-dimmer-bulb/ikea-xy-color-bulb/init.lua

```diff
@@ -66,36 +69,79 @@ end
 local function set_color_handler(driver, device, cmd)
   local hue = cmd.args.color.hue > 99 and 99 or cmd.args.color.hue
   local sat = cmd.args.color.saturation
-  local x, y, Y = utils.safe_hsv_to_xy(hue, sat)
-  store_xyY_values(device, x, y, Y)
-  switch_defaults.on(driver,device,cmd)
 
-  device:send(ColorControl.commands.MoveToColor(device, x, y, 0x0000))
 
-  device.thread:call_with_delay(2, query_device(device))
-end
-
-local function set_hue_handler(driver, device, cmd)
-  local sat = device:get_latest_state("main", capabilities.colorControl.ID, capabilities.colorControl.saturation.NAME)
-  local hue = cmd.args.hue > 99 and 99 or cmd.args.hue
   local x, y, Y = utils.safe_hsv_to_xy(hue, sat)
   store_xyY_values(device, x, y, Y)
-  switch_defaults.on(driver,device,cmd)
+  switch_defaults.on(driver, device, cmd)
 
   device:send(ColorControl.commands.MoveToColor(device, x, y, 0x0000))
 
+  device:set_field(TARGET_HUE, nil)
+  device:set_field(TARGET_SAT, nil)
   device.thread:call_with_delay(2, query_device(device))
 end
 
-local function set_saturation_handler(driver, device, cmd)
-  local hue = device:get_latest_state("main", capabilities.colorControl.ID, capabilities.colorControl.hue.NAME)
-  local x, y, Y = utils.safe_hsv_to_xy(hue, cmd.args.saturation)
-  store_xyY_values(device, x, y, Y)
-  switch_defaults.on(driver,device,cmd)
+local huesat_timer_callback = function(driver, device, cmd)
+  return function()
+    local hue = device:get_field(TARGET_HUE)
+    local sat = device:get_field(TARGET_SAT)
+    hue = hue ~= nil and hue or device.state_cache.main.colorControl.hue.value
```

Comment:
On initial join it's possible to have both `TARGET_HUE` and the state cache not have a value yet for the hue. (or vice versa with saturation)

Also I'm concerned that adding in the delay here as you've done hasn't caused any test cases to require adjustment.

Reply:
I'd expect they would need them, since you're adding at least a .2s delay that wasn't there before, no?

Reply:
That's a good point. I think there is something about the mock time handling that isn't quite right that was allowing the tests to pass before. I updated the tests with what I think should have been required and confirmed they are still passing. I think we need to talk more about the issues I was seeing but I don't think it needs to hold up the PR.